backend/graph/api/shorts.utils_test.go

```diff
@@ -0,0 +1,128 @@
+package graph
+
+import (
+	"github.com/bcc-code/bcc-media-platform/backend/common"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/guregu/null.v4"
+	"testing"
+)
+
+func Test_ShortsDeclumping(t *testing.T) {
+	type testCase struct {
+		declumpingTreshold int
+		input              []common.ShortIDWithMeta
+		expected           []uuid.UUID
+	}
+
+	testCases := []testCase{
+		{
+			input: []common.ShortIDWithMeta{
+				{
+					ID:              uuid.MustParse("00000000-0000-0000-0000-000000000001"),
+					ParentEpisodeID: null.IntFrom(1),
+				},
+				{
+					ID:              uuid.MustParse("00000000-0000-0000-0000-000000000002"),
+					ParentEpisodeID: null.IntFrom(1),
+				},
+				{
+					ID:              uuid.MustParse("00000000-0000-0000-0000-000000000003"),
+					ParentEpisodeID: null.IntFrom(2),
+				},
+				{
+					ID:              uuid.MustParse("00000000-0000-0000-0000-000000000004"),
+					ParentEpisodeID: null.IntFrom(2),
+				},
+			},
+			expected: []uuid.UUID{
+				uuid.MustParse("00000000-0000-0000-0000-000000000001"),
+				uuid.MustParse("00000000-0000-0000-0000-000000000003"),
+			},
+			declumpingTreshold: 3,
+		},
+		{
+			input: []common.ShortIDWithMeta{
+				{
+					ID:              uuid.MustParse("00000000-0000-0000-0000-000000000001"),
+					ParentEpisodeID: null.IntFrom(1),
+				},
+				{
+					ID:              uuid.MustParse("00000000-0000-0000-0000-000000000002"),
+					ParentEpisodeID: null.IntFrom(1),
+				},
+				{
+					ID:              uuid.MustParse("00000000-0000-0000-0000-000000000003"),
+					ParentEpisodeID: null.IntFrom(1),
+				},
+				{
+					ID:              uuid.MustParse("00000000-0000-0000-0000-000000000004"),
+					ParentEpisodeID: null.IntFrom(1),
+				},
+			},
+			expected: []uuid.UUID{
+				uuid.MustParse("00000000-0000-0000-0000-000000000001"),
+				uuid.MustParse("00000000-0000-0000-0000-000000000004"),
+			},
+			declumpingTreshold: 3,
+		},
```

Comment:
What would you guys expect as the result here? Is it what I made or would you expect only `00000000-0000-0000-0000-000000000001` to be in the result?

Reply:
I think it makes sense if we think about it like the threshold is how many until the next one, instead of how many between this and the next one.